### ISSUE TITLE
USE_MD5_Go in dynamic is never used

### ISSUE THREAD

**Issue by frank-dittrich** [magnumripper/JohnTheRipper#1298]
May be the code snippets were once correct, but aren't anymore after some code restructuring.

```
$ git grep -B 5 -n USE_MD5_Go
dynamic_fmt.c-150-#if ARCH_LITTLE_ENDIAN
dynamic_fmt.c-151-// MD5_go is SUPER slow on big endian. In the case of bigendian, we simply
dynamic_fmt.c-152-// fall back, and use OpenSSL MD5 calls, which are usually MUCH faster.
dynamic_fmt.c-153-#ifndef _OPENMP
dynamic_fmt.c-154-// NOTE, MD5_go is NOT thread safe.
dynamic_fmt.c:155:#define USE_MD5_Go
--
dynamic_types.h-251-#define ALGORITHM_NAME_X86_TGR "32/"ARCH_BITS_STR " sph_tiger"
dynamic_types.h-252-
dynamic_types.h-253-#define ALGORITHM_NAME_RIPEMD     "32/"ARCH_BITS_STR " sph_ripmd"
dynamic_types.h-254-#define ALGORITHM_NAME_X86_RIPEMD "32/"ARCH_BITS_STR " sph_ripmd"
dynamic_types.h-255-
dynamic_types.h:256:#ifdef USE_MD5_Go
--
dynamic_types.h-292-//static MD5_OUT tmpOut;
dynamic_types.h-293-#define DoMD5a(A,L,C) do{MD5_body(A->x1.w,C->x1.w);MD5_swap(C->x1.w,C->x1.w,4);}while(0)
dynamic_types.h-294-//#define DoMD5a2(A,L,C,D) do{MD5_body(A->x1.w,tmpOut.x1.w);MD5_swap(tmpOut.x1.w,tmpOut.x1.w,4);memcpy(&(C->x1.b[D[0]]),tmpOut.x1.b,16);}while(0)
dynamic_types.h-295-#endif
dynamic_types.h-296-#endif
dynamic_types.h:297:#else // !USE_MD5_Go
```

So, dynamic_fmt.c line 155 is the only place where USE_MD5_Go can be defined.
dynamic_types.h has the only code that depends on USE_MD5_Go.
But:
```
$ git grep -n dynamic_types.h
Makefile.in:336:dynamic_big_crypt.o:    dynamic_big_crypt.c arch.h openssl_local_overrides.h misc.h common.h formats.h sha.h sha2.h md5.h md4.h stdint.h gost.h sph_haval.h sph_ripemd.h sph_tiger.h sph_whirlpool.h dynamic.h johnswap.h sse-intrinsics.h dynamic_types.h memdbg.h
Makefile.in:338:dynamic_fmt.o:  dynamic_fmt.c arch.h sse-intrinsics.h misc.h common.h formats.h params.h md5.h md4.h dynamic.h options.h config.h sha.h gost.h memory.h unicode.h johnswap.h pkzip.h aligned.h fake_salts.h dynamic_types.h memdbg.h
dynamic_big_crypt.c:63:#include "dynamic_types.h"
dynamic_fmt.c:131:#include "dynamic_types.h"
```
dynamic_fmt.c includes dynamic_types.h in line 131, i.e., before USE_MD5_Go might be defined.
So, apparently, dynamic doesn't use MD5_Go() anymore.

Can we just get rid of this in dynamic?

**Comment by jfoug** [comment]
I am not sure that MD5_go is used by any format at all any more.   MD5_go was the first code used in dynamic (other than oSSL), as it used to be faster 5 years ago when mingw had a horrible performing oSSL.  But years ago MD5_body (1x and 2x), and thread safe version.   I will look into this to make 100% sure.

**Comment by jfoug** [comment]
I have simply removed it. I think at one time, it was faster than oSSL, and it was the first thing I used.  Later I started using MD5_body (same code as md5crypt uses), and found it better all around.  The #defines were left in, but as was pointed out, can not be used.  The MD5_go was the original reason why we had to have a larger buffer (135 bytes, I think), because that code would spill over the md5 buffer.  But now we have larger buffers, due to doing multi buffer crypts in dyna.

3c5c283 removes the MD5_Go macros and code.
